nats.go

```diff
@@ -3884,15 +3884,22 @@ func (nc *Conn) Stats() Statistics {
 	// Stats are updated either under connection's mu or subsMu mutexes.
 	// Lock both to safely get them.
 	nc.mu.Lock()
-	nc.subsMu.RLock()
+	// Note: It would be more logical to get subsMu's read lock here since we are
+	// "reading" the stats. However, in processMsg() we don't lock the connection
+	// (unless we detect a slow consumer) but we do get subsMu read lock to fetch
+	// the subscription the incoming message is for. We do update InMsgs/Bytes under
+	// that read lock. Swithching processMsg()'s subsMu to write lock would decrease
```

Comment:
Spelling on Switching.

Reply:
We do an update under a read lock? That sounds wrong.

Reply:
Replaced all that with updates of InMsgs/InBytes in processMsg() using atomic add and atomic load in Stats().

Reply:
How does that affect performance? I think that or a simple lock depending on which performs better.

Reply:
Getting subsMu.Lock() (instead of RLock) dropped perf by 5%. Replacing with atomic dropped by 2%, this is why I changed.
Note that although unconventional, the original approach of getting the write lock in Stats() as opposed to processMsg() was not affecting performance (obviously since there was no change in processMsg()).

Reply:
I like perf more than anyone, but this is the correct way IMO. Thanks.